### DISCUSSION TITLE
Cannot pod install firestore package on Apple Silicon M1

### DISCUSSION THREAD

**Discussion by anhnch** [invertase/react-native-firebase#6630] [Q&A]
I'm not sure this is cocoapod or firebase issue. But I'm sure it occurs on Macbook chip M1, because everything is fine on my other non-M1 devices.

First I add react native firebase packages with yarn. Then I run pod install
All packages (app, analytics, storage, remote-config, auth, functions, messaging, crashlytics) work fine except this one @react-native-firebase/firestore
```
yarn add @react-native-firebase/firestore <= OK
pod install <= FAILED
```
The output is weird, I can't find any clue. No explicit error, it just spits out bunch of text that is way too long and not useful to post here. But there's one detail:

```
### Error

JSON::ParserError - 859: unexpected token at '{
  "name": "BoringSSL-GRPC",
  "version": "0.0.24",
  "summary": "BoringSSL is a fork of OpenSSL that is designed to meet Google's needs.",
  "description": "BoringSSL is a fork of OpenSSL that is designed to meet Google's needs.\n\nAlthough BoringSSL is an open source project, it is not intended for general use, as OpenSSL is.\nWe don't recommend that third parties depend upon it. Doing so is likely to be frustrating\nbecause there are no guarantees of API stability. Only the latest version of this pod is\nsupported, and every new version is a new major version.\n\nWe update Google libraries and programs that use BoringSSL as needed when deciding to make API\nchanges. This allows us to mostly avoid compromises in the name of compatibility. It works for\nus, but it may not work for you.\n\nAs a Cocoapods pod, it has the advantage over OpenSSL's pods that the library doesn't need to\nbe precompiled. This eliminates the 10 - 20 minutes of wait the first time a user does \"pod\ninstall\", lets it be used as a dynamic framework (pending solution of Cocoapods' issue #4605),\nand works with bitcode automatically. It's also thought to be smaller than OpenSSL (which takes\n1MB - 2MB per ARM architecture), but we don't have specific numbers yet.\n\nBoringSSL arose because Google used OpenSSL for many years in various ways and, over time, built\nup a large number of patches that were maintained while tracking upstream OpenSSL. As Google's\nproduct portfolio became more complex, more copies of OpenSSL sprung up and the effort involved\nin maintaining all these patches in multiple places was growing steadily.\n\nCurrently BoringSSL is the SSL library in Chrome/Chromium, Android (but it's not part of the\nNDK) and a number of other apps/programs.",
  "homepage": "https://github.com/google/boringssl",
  "license": {
    "type": "Mixed",
    "file": "LICENSE"
  },
```
There must be something wrong with BorringSSL, somehow somewhere during building firestore. 
I found a walkaround from Mike's post https://github.com/invertase/react-native-firebase/issues/2564#issuecomment-861586844 

Downloading a pre-compiled solves this, for now! I'm not sure this is a good fix.

More info here: https://github.com/invertase/firestore-ios-sdk-frameworks 

I hope you guys find this helpful

**Comment by mikehardy** [comment]
This is not an issue in this module, this will be upstream - but even "upstream upstream", as BoringSSL is a transitive dependency of firebase-ios-sdk frameworks. 

https://github.com/mikehardy/rnfbdemo/blob/main/make-demo.sh is my standard build/integration demonstrator, I would try that to prove that an automated install is not working, then I would search firebase-ios-sdk

I run my build/integration demonstrator on M1 machines frequently and it works, so this is not an M1-specific issue I think, it is perhaps a machine-specific issue

**Comment by anhnch** [comment]
Please check the [log.txt](https://github.com/invertase/react-native-firebase/files/9869827/log.txt)
